### ISSUE TITLE
Supervised and Unsupervised Learning Support

### ISSUE THREAD

**Issue by donamin** [Unity-Technologies/ml-agents#4257]
I am developing a ML project using ml-agents. One issue that I'm struggling with right now, is that the framework has been solely developed for reinforcement learning. But what about other types of learning, i.e., supervised and unsupervised learning? For example, part of my project requires an out-of-the-box nn trained using some offline data. I believe with all the cool features in ml-agents, it would be an easy task to add support for other types of learning.

**Comment by vincentpierre** [comment]
Hi,
You are correct, ML-Agents is built around reinforcement learning. There are other tools we are working on for supervised learning like the [Unity Perception SDK](https://blogs.unity3d.com/2020/06/10/use-unitys-perception-tools-to-generate-and-analyze-synthetic-data-at-scale-to-train-your-ml-models/). If you want to use ML-Agents and do supervised learning, you can collect data with a reinforcement learning loop and then save / reuse the data to train a supervised learning model.
I will mention your interest for supervised and unsupervised learning to the team, do you have specific use cases in mind ?

**Comment by donamin** [comment]
Thanks for the reply.

Well, the use cases can be very generic. For example, RL systems (such as [this one](https://www.youtube.com/watch?v=Rzj3k3yerDk&feature=youtu.be)) can have several supervised or unsupervised components for helping the decision making process. I'm particuarly interested in using generative models in Unity. For that, one needs a library that allows one to design the neural network architecture and choose the loss function themselves.

**Comment by vincentpierre** [comment]
Ha, I thought you wanted to not do RL and only supervised.
In our trainers, we use supervised learning in the [behavioral cloning module](https://github.com/Unity-Technologies/ml-agents/blob/release_4/docs/Training-Configuration-File.md#behavioral-cloning) and GANs in the [GAIL reward module](https://github.com/Unity-Technologies/ml-agents/blob/release_4/docs/Training-Configuration-File.md#gail-intrinsic-reward). 
If you think of other generic RL components that can help learning, let us know. We currently do not have a solution for animations as we are focused on generating behaviors.
